server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/EnqueuedMailsDaoUtil.java

```diff
@@ -177,16 +177,15 @@ private static MailAddress toMailAddress(String rawValue) {
     }
 
     static ImmutableMap<String, ByteBuffer> toRawAttributeMap(Mail mail) {
-        return Iterators.toStream(mail.getAttributeNames())
-            .map(name -> Pair.of(name, mail.getAttribute(name)))
-            .map(pair -> Pair.of(pair.getLeft(), toByteBuffer(pair.getRight())))
+        return mail.attributes()
+            .map(attribute -> Pair.of(attribute.getName().asString(), toByteBuffer(attribute.getValue())))
             .collect(ImmutableMap.toImmutableMap(Pair::getLeft, Pair::getRight));
     }
 
-    private static ByteBuffer toByteBuffer(Serializable serializable) {
+    private static ByteBuffer toByteBuffer(AttributeValue<?> attributeValue) {
         try {
             ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            new ObjectOutputStream(outputStream).writeObject(serializable);
+            new ObjectOutputStream(outputStream).writeObject(attributeValue.toJson().toString());
```

Comment:
:red_circle:  isn't it `return ByteBuffer.wrap(attributeValue.toJson().toString().getBytes(StandardCharsets.UTF-8))` ?

We don't need that intermediate `Serialized Object` representation which only adds security vulnerability and complexity.